[NNAPI] Add support for unsqueeze, cat, and mean

dreiss opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #54701 [NNAPI] Initial flexible size support
* #54700 [NNAPI] Create get_next_operand_id
* #54699 [NNAPI] Add fixed-size assertions
* #54698 [NNAPI] Rename local variable
* #54697 [NNAPI] Use code generation to better support list input/output
* #54696 [NNAPI] Use array instead of struct for serializing ints
* #48925 Fix NNAPI for internal fbcode build
* #48812 [NNAPI] Handle binary ops combining NHWC+NCHW in some cases
* **#48811 [NNAPI] Add support for unsqueeze, cat, and mean**
* #47521 [NNAPI] Add unit test
* #54695 [NNAPI] Update support for Linear
* #47520 [NNAPI] Fix hardtanh
* #47519 [NNAPI] Fix MUL op
* #47518 [NNAPI] Remove solid weights support
* #47517 [NNAPI] Fix models with no weights
* #47407 Make bundled inputs work with quantized zero inputs

Test Plan:
Unit tests.

Differential Revision: [D25317936](https://our.internmc.facebook.com/intern/diff/D25317936)

dreiss committed:
[NNAPI] Add support for unsqueeze, cat, and mean

Test Plan:
Unit tests.

[ghstack-poisoned]

dreiss committed:
Update on "[NNAPI] Add support for unsqueeze, cat, and mean"


Test Plan:
Unit tests.

Differential Revision: [D25317936](https://our.internmc.facebook.com/intern/diff/D25317936)

[ghstack-poisoned]

axitkhurana commented on torch/backends/_nnapi/serializer.py:
```diff
@@ -698,6 +751,94 @@ def add_reshape(self, node):
+
+        out_oper = in_oper._replace(shape=out_shape, dim_order=out_dim_order)
+
+        inputs = [None] * 3
```
Any reason for not doing `inputs = [in_id, ....]` here? Same for outputs

dreiss committed:
Update on "[NNAPI] Add support for unsqueeze, cat, and mean"


Test Plan:
Unit tests.

Differential Revision: [D25317936](https://our.internmc.facebook.com/intern/diff/D25317936)

[ghstack-poisoned]

dreiss committed:
Update on "[NNAPI] Add support for unsqueeze, cat, and mean"


Test Plan:
Unit tests.

Differential Revision: [D25317936](https://our.internmc.facebook.com/intern/diff/D25317936)

[ghstack-poisoned]

dreiss committed:
Update on "[NNAPI] Add support for unsqueeze, cat, and mean"


Test Plan:
Unit tests.

Differential Revision: [D25317936](https://our.internmc.facebook.com/intern/diff/D25317936)

[ghstack-poisoned]

dreiss committed:
Update on "[NNAPI] Add support for unsqueeze, cat, and mean"


Test Plan:
Unit tests.

Differential Revision: [D25317936](https://our.internmc.facebook.com/intern/diff/D25317936)

[ghstack-poisoned]

dreiss committed:
Update on "[NNAPI] Add support for unsqueeze, cat, and mean"


Test Plan:
Unit tests.

Differential Revision: [D25317936](https://our.internmc.facebook.com/intern/diff/D25317936)

[ghstack-poisoned]

dreiss committed:
Update on "[NNAPI] Add support for unsqueeze, cat, and mean"


Test Plan:
Unit tests.

Differential Revision: [D25317936](https://our.internmc.facebook.com/intern/diff/D25317936)

[ghstack-poisoned]

dreiss committed:
Update on "[NNAPI] Add support for unsqueeze, cat, and mean"


Test Plan:
Unit tests.

Differential Revision: [D25317936](https://our.internmc.facebook.com/intern/diff/D25317936)

[ghstack-poisoned]

dreiss committed:
Update on "[NNAPI] Add support for unsqueeze, cat, and mean"


Test Plan:
Unit tests.

Differential Revision: [D25317936](https://our.internmc.facebook.com/intern/diff/D25317936)

[ghstack-poisoned]

dreiss committed:
Update on "[NNAPI] Add support for unsqueeze, cat, and mean"


Test Plan:
Unit tests.

Differential Revision: [D25317936](https://our.internmc.facebook.com/intern/diff/D25317936)

[ghstack-poisoned]
